lightningd/peer_htlcs.c

```diff
@@ -929,9 +929,29 @@ static bool htlc_accepted_hook_deserialize(struct htlc_accepted_hook_payload *re
 
 	if (json_tok_streq(buffer, resulttok, "fail")) {
 		u8 *failmsg;
-		const jsmntok_t *failmsgtok, *failcodetok;
+		const jsmntok_t *failoniontok, *failmsgtok, *failcodetok;
 
+		failoniontok = json_get_member(buffer, toks, "failure_onion");
 		failmsgtok = json_get_member(buffer, toks, "failure_message");
+
+		if (failoniontok) {
+			failonion = json_tok_bin_from_hex(tmpctx, buffer,
```

Comment:
@fiatjaf the memory leak was caused by this line

```original
			failonion = json_tok_bin_from_hex(NULL, buffer,
```

By allocating it to `NULL` this means you're planning to manually clean it up; instead we use the magic allocation context `tmpctx`, which gets cleaned up, via magic.